api/nameguard/provider.py

```diff
@@ -3,10 +3,16 @@
 
 from nameguard.logging import logger
 from nameguard.exceptions import ProviderUnavailable
+from nameguard.models import NetworkName
 
+provider_uris = {
+    NetworkName.MAINNET: os.environ.get('PROVIDER_URI_MAINNET'),
+    NetworkName.SEPOLIA: os.environ.get('PROVIDER_URI_SEPOLIA'),
+}
 
-async def get_nft_metadata(contract_address: str, token_id: str) -> dict:
-    url = f"{os.environ.get('PROVIDER_URI_MAINNET')}/getNFTMetadata?contractAddress={contract_address}&tokenId={token_id}&refreshCache=false"
+
+async def get_nft_metadata(network_name, contract_address: str, token_id: str) -> dict:
+    url = f'{provider_uris[network_name]}/getNFTMetadata?contractAddress={contract_address}&tokenId={token_id}&refreshCache=false'
```

Comment:
`getNFTMetadata` isn't a request that works across providers. This API is specific to Alchemy.

In the context of Ethereum, a "provider" is a system running a standard Ethereum API endpoint. Alchemy extends this with some proprietary APIs including `getNFTMetadata.

Therefore we should have separate / distinct environment variables for the Alchemy API endpoint vs the general Ethereum provider API endpoint.

We use the Alchemy API endpoint only for these NFT Metadata lookups.

We use the general Ethereum provider API endpoint for other lookups, such as what is the primary name of a given address, etc.

Please let me know if any questions :+1: